courses/models.py

```diff
@@ -0,0 +1,52 @@
+from django.core.urlresolvers import reverse
+from django.contrib.auth.models import User
+from django.db import models
+from jsonfield import JSONField
+from uuidfield import UUIDField
+
+class Course(models.Model):
+    """
+    An edX course which is purchaseable.
+    """
+    uuid = UUIDField(auto=True)
+    title = models.CharField(max_length=255)
+    author_name = models.CharField(max_length=255)
+    overview = models.TextField()
+    description = models.TextField()
+    video_url = models.URLField()
+
+    edx_instance = models.CharField(max_length=255)
+    instructors = models.ManyToManyField(User)
+    live = models.BooleanField(default=False)
+
+    price_per_seat_cents = models.IntegerField(null=True, blank=True, help_text="Cost of the whole course per seat in cents")
+
+    deleted = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True, blank=True)
+    updated_at = models.DateTimeField(auto_now=True, blank=True)
+
+    def __unicode__(self):
+        return self.title
+
+    @property
+    def url(self):
+        return reverse('course-detail', args=(self.uuid,))
+
+
+class Module(models.Model):
+    """
+    Represents a purchasable 'chapter' of a course.
+    """
+    uuid = UUIDField(auto=True)
+    course = models.ForeignKey(Course)
+    title = models.CharField(max_length=255)
+    subchapters = JSONField(default="[]") # Array of strings.
+    locator_id = models.CharField(max_length=255)
+    price_per_seat_cents = models.IntegerField(blank=True, null=True)
```

Comment:
The business logic for this will be in the ecommerce piece, I assume, but
you can toggle between "buy the course" or "buy specific modules". The
course price isn't necessarily the sum of chapters.

On Thu, Nov 12, 2015 at 2:17 PM, George Schneeloch <notifications@github.com
> wrote:

> In courses/models.py
> <https://github.com/mitodl/ccxcon/pull/7#discussion_r44701077>:
>
> > +
> > +    @property
> > +    def url(self):
> > +        return reverse('course-detail', args=(self.uuid,))
> > +
> > +
> > +class Module(models.Model):
> > +    """
> > +    Represents a purchasable 'chapter' of a course.
> > +    """
> > +    uuid = UUIDField(auto=True)
> > +    course = models.ForeignKey(Course)
> > +    title = models.CharField(max_length=255)
> > +    subchapters = JSONField(default="[]") # Array of strings.
> > +    locator_id = models.CharField(max_length=255)
> > +    price_per_seat_cents = models.IntegerField(blank=True, null=True)
>
> Does this take precedence over the course's price_per_seat_cents if
> specified, or do they refer to two different things?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mitodl/ccxcon/pull/7/files#r44701077>.
>